.github/workflows/cxx.yml

```
@@ -0,0 +1,247 @@
+          # FIXME: Can't test clang+Linux because DMD assume g++
+          - { os: ubuntu-16.04, target: clang-9.0.0 }
+          - { os: ubuntu-16.04, target: clang-8.0.0 }
+          - { os: ubuntu-16.04, target: clang-7.0.0 }
+          - { os: ubuntu-16.04, target: clang-6.0.0 }
+          - { os: ubuntu-16.04, target: clang-5.0.2 }
+          - { os: ubuntu-16.04, target: clang-4.0.0 }
+          - { os: ubuntu-16.04, target: clang-3.9.0 }
+          # OSX only supports clang
+          - { os: macOS-10.15, target: g++-9 }
+          - { os: macOS-10.15, target: g++-8 }
+          - { os: macOS-10.15, target: g++-7 }
+          - { os: macOS-10.15, target: g++-6 }
+          - { os: macOS-10.15, target: g++-5 }
+          - { os: macOS-10.15, target: msvc-2019 }
+          - { os: macOS-10.15, target: msvc-2017 }
+          - { os: macOS-10.15, target: msvc-2015 }
+          - { os: macOS-10.15, target: msvc-2013 }
+          # We don't test g++ on Windows as DMD only mangles for MSVC
+          - { os: windows-2019, target: g++-9 }
+          - { os: windows-2019, target: g++-8 }
+          - { os: windows-2019, target: g++-7 }
+          - { os: windows-2019, target: g++-6 }
+          - { os: windows-2019, target: g++-5 }
+
+          # TODO: Implement support for clang and MSVC on Windows
+          # Currently those are still being run by the auto-tester
+          - os: windows-2019
+
+        include:
+          # Clang boilerplate
+          - { target: clang-9.0.0, compiler: clang, cxx-version: 9.0.0 }
+          - { target: clang-8.0.0, compiler: clang, cxx-version: 8.0.0 }
+          - { target: clang-7.0.0, compiler: clang, cxx-version: 7.0.0 }
+          - { target: clang-6.0.0, compiler: clang, cxx-version: 6.0.0 }
+          - { target: clang-5.0.2, compiler: clang, cxx-version: 5.0.2 }
+          - { target: clang-4.0.0, compiler: clang, cxx-version: 4.0.0 }
+          - { target: clang-3.9.0, compiler: clang, cxx-version: 3.9.0 }
+          # g++ boilerplace
+          - { target: g++-9, compiler: g++, cxx-version: 9.3.0 }
+          - { target: g++-8, compiler: g++, cxx-version: 8.4.0 }
+          - { target: g++-7, compiler: g++, cxx-version: 7.5.0 }
+          - { target: g++-6, compiler: g++, cxx-version: 6.5.0 }
+          - { target: g++-5, compiler: g++, cxx-version: 5.5.0 }
+          # Platform boilerplate
+          - { os: ubuntu-16.04, arch: x86_64-linux-gnu-ubuntu-16.04 }
+          - { os: macOS-10.15,  arch: x86_64-apple-darwin }
+          # Clang 9.0.0 have a different arch for OSX
+          - { os: macOS-10.15, target: clang-9.0.0, arch: x86_64-darwin-apple }
+
+    runs-on: ${{ matrix.os }}
+    steps:
+
+    ########################################
+    #    Setting up the host D compiler    #
+    ########################################
+    - name: Prepare compiler
+      uses: mihails-strasuns/setup-dlang@v0.5.0
+      with:
+          compiler: dmd-2.091.0
```

```suggestion
          compiler: ${{ matrix.host_dc }}
```

Make this a variable, by adding it as a matrix dimension, but with just a single item:

```yml
matrix:
  host_dc: [ dmd-2.091.0 ]
  // ...
```

---

I considered doing this at first, but decided against it. One of the reason to do so is that it shows up in the matrix, thus leading to longer name (and the host compiler information is not that important). The other reason is that I didn't want to add a level of indirection or mislead people into believing more dimensions should be added.

Is there any reason to make it part of the matrix when we only want 1 entry ?

---

The biggest reason for me is to make it easy change configuration by extracting things like that as variables in one central place. I don't know if using a matrix of the best way to achieve this. In most other CI scripts in the dlang repos, test inputs like the D compiler are parameterized.